documentation/c01-whatis.sil

```diff
@@ -82,11 +82,105 @@ SILE does not benefit from the large ecosystem and community that has grown up a
+	Pandoc support allows some other formats to be converted and imported, but not worked with natively.}
+
+\item{\strong{Scripting:}
+	SILE exposes (and indeed is actually mostly implemented in) Lua, an industry standard embedded scripting language with an already large ecosystem of code and packages.
+	Lua’s syntax and features are well known and documented.
+	Besides being used in other Typesetting applications\footnote{%
+		SILE was initially implemented in JavaScript, then ported to Lua.
+		The thought behind the switch to Lua was to gain advantage of its use in LuaLaTeX and other typesetting applications.
+		In practice, using the same language as other typesetting systems has not really provided much of the hoped for opportunities for code re-use.
+		It turns out algorithms and expertise themselves are more relevant than the language they happen to be written in and LuaLaTeX specific libraries are not very flexible.}
+	it can be found embedded in web servers, game engines, code editors, and a myriad of other applications.
+	Typst’s approach to scripting was to invent their own language from scratch.
+	The language they invented is clean, logical, and robust—but also unique and, to some extent, constraining.}
+
+\item{\strong{Security model:}
+	Typst attempts to make the renderer safe to run on arbitrary input.
+	This is a noble goal and serves a purpose for some use cases, but by definition it introduces significant limitations.
+	The scripting language is deliberately sandboxed and not allowed direct access to arbitrary disk I/O, network, or system processes.
+	SILE on the other hand makes the assumption that the user, including whatever scripting they may embed in a document or project, should have complete freedom to do anything the user wants.
+	As such document rendering is not sandboxed at all and scripting has complete control to do anything the user of the system is permitted—to include spawning processes, accessing the network, and file system access.}
+
+\item{\strong{Adaptability:}
+	Typst’s core engine is written in Rust, and very specific APIs are provided to the user via the custom scripting interface.
+	This interface is well documented and really quite versatile considering the design constraint, but it does not allow any arbitrary part of the engine to be accessed or changed.
+	In SILE’s case 100\% of the internals are exposed via Lua and/or FFI and can be inspected or overridden by the user.\footnote{%
+		In addition to the native Lua interface, users may extend SILE with modules written in Rust, C, or any other language with an FFI interface or any executable that can be spawned in a process.
+		In other words almost anything can be integrated.}
+	This includes the parts documented to be user-facing API surfaces; but also the core modules and other bits the SILE developers generally consider as private APIs.
```

Comment:
"... the SILE developers generally consider as private APIs" -- We all know it's not a clear definition, at the risk of _anyone_'s code being called "upside down", or claimed to be doing things"backwards" etc. :grin: